lib/features/push_notification/presentation/listener/email_change_listener.dart

```diff
@@ -263,6 +267,25 @@ class EmailChangeListener extends ChangeListener {
     required List<PresentationEmail> emailList
   }) async {
     log('EmailChangeListener::_handleLocalPushNotification(): EMAIL_LENGTH = ${emailList.length}');
+
+    if (PlatformInfo.isAndroid) {
+      final notificationSettingEnabled = (await _getAppNotificationSettingCacheInteractor?.execute().last)
```

Comment:
We should check it once in the constructor, instead of multiple times every time we show a push notification.

Reply:
What if the permission is changed after the class is initiated? `EmailChangeListener` is a Singleton.

Reply:
We still have to request it but not in the push notification handler function. We should handle it in inject dependencies constructors.

Reply:
Please clarify

Reply:
Will update